### ISSUE TITLE
Angular component which references its own type and decorated by class decorator is broken when target es5 + enableIvy=true

### ISSUE THREAD

**Issue by ratoaq2** [angular/angular#39509]
# 🐞 bug report

### Affected Package
The issue is related to angular ivy compiler

### Is this a regression?
Yes, the exact use case works without issues in Angular 8.0.x (without ivy).

### Description
Given a normal angular component which references its own type in component metadata, and it's also decorated with an additional class decorator:
```
import { Component, InjectionToken, Input } from "@angular/core";

const BOUND_COMPONENT = new InjectionToken<any>("BoundComponent");

export function Bind() {
  return function(target: any) {
    console.log("Bound", target);
  };
}

@Bind()
@Component({
  selector: "hello",
  template: `
    <h1>Hello {{ name }}!</h1>
  `,
  providers: [{ provide: BOUND_COMPONENT, useExisting: HelloComponent }],
  styles: [
    `
      h1 {
        font-family: Lato;
      }
    `
  ]
})
export class HelloComponent {
  @Input() name: string;
}

will give an runtime error as you can see in stackblitz
```

## 🔬 Minimal Reproduction
https://stackblitz.com/edit/angular-ivy-ucqpbt

## 🔥 Exception or Error
<pre><code>
Error in /~/src/app/hello.component.ts
HelloComponent_1 is not defined
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>
ng version

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 10.0.8
Node: 12.18.4
OS: win32 x64

Angular: 10.0.5
... animations, common, compiler, compiler-cli, core, forms
... router
Ivy Workspace: <error>

Package                             Version
-------------------------------------------------------------
@angular-devkit/architect           0.1000.8
@angular-devkit/build-angular       0.1000.8
@angular-devkit/build-ng-packagr    0.1000.8
@angular-devkit/build-optimizer     0.1000.8
@angular-devkit/build-webpack       0.1000.8
@angular-devkit/core                10.0.8
@angular-devkit/schematics          10.0.8
@angular/cdk                        10.1.0
@angular/cli                        10.0.8
@angular/flex-layout                10.0.0-beta.32
@angular/http                       8.0.0-beta.10
@angular/language-service           10.2.0
@angular/localize                   10.2.0
@angular/material                   10.1.0
@angular/material-moment-adapter    10.1.0
@angular/platform-browser           10.2.0
@angular/platform-browser-dynamic   10.2.0
@angular/upgrade                    10.0.14
@ngtools/webpack                    10.0.8
@schematics/angular                 10.0.8
@schematics/update                  0.1000.8
ng-packagr                          10.1.2
rxjs                                6.6.0
typescript                          3.9.7
webpack                             4.43.0
</code></pre>

**Anything else relevant?**
Nope

**Comment by ratoaq2** [comment]
Just realized that this issue was never fixed in 10.x version, only 11.x.
Would it be possible to merge this fix in 10.x version as well? A more recent issue that I reported was successfully merged in 10.2.x.

Thanks

**Comment by petebacondarwin** [comment]
Yes, I would have expected it to land on `10.2.x` but it was marked as targeting `rc` which resulted it only going in `master` and `11.0.x`. I'll check with @josephperrott and @crisbeto ...

**Comment by petebacondarwin** [comment]
Hi @ratoaq2 - I checked with the team. While this fix could have been merged into 10.2.x, that branch is now not the active patch branch. It is our standard policy to only backport security and catastrophic bugs to earlier branches. Please can I ask you to attempt an update to 11.0.x to receive the benefit of this fix?

**Comment by ratoaq2** [comment]
Hi @petebacondarwin 

Unfortunately that's not an issue for us.
Our application belongs to a big enterprise ecosystem and we can't just move to Angular 11.
Isn't it possible to get this bug backported?

Thanks for the support

**Comment by petebacondarwin** [comment]
Unfortunately, even if this commit was merged to the 10.2.x branch, there is no planned release of that branch at this time. I would strongly advise pushing for your enterprise to update - which is good practice anyway since any other future bugs that are found would not be released on 10.2.x either.

A workaround for you could be to apply the patch locally to the [src/ngtsc/annotations/src/metadata.js](https://unpkg.com/browse/@angular/compiler-cli@10.2.3/src/ngtsc/annotations/src/metadata.js) as a postinstall step.
